ingest/modules/normedPublication.ts

```diff
@@ -1,14 +1,191 @@
-// the normalized simple form of a publication across all sources
-interface NormedPublication {
-  searchPerson?: NormedPerson,
-  abstract?: string,
-  title: string,
-  journalTitle: string,
-  journalIssn?: string,
-  journalEIssn?: string,
-  doi: string,
-  publicationDate: string,
-  datasourceName: string,
-  sourceId?: string,
+import _ from 'lodash'
+import fs from 'fs'
+import { getDateObject } from '../units/dateRange'
+import { command as loadCsv } from '../units/loadCsv'
+import { command as writeCsv} from '../units/writeCsv'
+
+export default class NormedPublication {
+  // ------ begin declare properties used when using NormedPublication like an interface
+
+  // the normalized simple form of a publication across all sources
+  searchPerson?: NormedPerson
+  abstract?: string
+  title: string
+  journalTitle: string
+  journalIssn?: string
+  journalEIssn?: string
+  doi: string
+  publicationDate: string
+  datasourceName: string
+  sourceId?: string
   sourceMetadata?: Object
+  // ------- end declare properties used when using NormedPublication like an interface
+
+  // begin declaring static utility methods for NormedPublication objects
+  //-----------------------------------------------------------------------
+  /**
+   * 
+   * @param csvPath the path the csv file containing the publications to be loaded
+   * 
+   * @param columnNameMap a map of column names in the csv to harvestset property names, if not defined uses default path from configuration
+   * 
+   * @returns object with array of raw publication set as well as hash of doi to index of corresponding publication in array
+   */
+  public static async loadFromCSV (csvPath: string): Promise<NormedPublication[]> {
+    console.log(`Loading Papers from path: ${csvPath}`)
+    // ingest list of DOI's from CSV and relevant center author name
+    try {
+
+      const objectToCSVMap = NormedPublication.loadNormedPublicationObjectToCSVMap()
+
+      const authorPapers: any = await loadCsv({
+        path: csvPath,
+        lowerCaseColumns: true
+      })
+
+      let sourceName = undefined
+      if (authorPapers.length > 0){
+        sourceName = authorPapers[0]['sourcename']
+      }
+
+      return _.map(authorPapers, (paper) => {
+        return NormedPublication.getNormedPublicationObjectFromCSVRow(paper, objectToCSVMap)
+      })
+    } catch (error){
+      console.log(`Error on paper load for path ${csvPath}, error: ${error}`)
+      return undefined
```

Comment:
This seems like we'd want to throw an error instead of returning an undefined which will likely raise a more opaque error somewhere up the stack